flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1766,7 +1766,8 @@ class ScalarExprLowering {
                                                        typeParams, extents);
     auto *bldr = &converter.getFirOpBuilder();
     // TODO: call finalizer if needed.
-    stmtCtx.attachCleanup([=]() { bldr->create<fir::FreeMemOp>(loc, temp); });
+    stmtCtx.attachCleanup(
+        [bldr, loc, temp]() { bldr->create<fir::FreeMemOp>(loc, temp); });
```

Comment:
What is the rational behind the lambda capture change, should we change to explicit captures as a general rule ?

Reply:
The rationale here is that we don't really want to capture `this` in cleanups at all, because it will be a dangling reference. Omitting the default specifier will give a compiler error. This instance is trivial and it is clearly not capturing `this`, but some cleanups are a bit more complicated.